File: 0543-diameter-of-binary-tree/0543-diameter-of-binary-tree.cpp
class Solution {
    int helper(TreeNode* root,int& maxdia) {
        if (root == NULL)
            return 0;
        int dia = levels(root->left) + levels(root->right);
        helper(root->right,maxdia);
        helper(root->left,maxdia);
        maxdia = max(maxdia, dia);
        return maxdia;
    }
    int levels(TreeNode* root) {
        if (root == NULL)
            return 0;
        return 1 + max(levels(root->right), levels(root->left));
    }

public:
    int diameterOfBinaryTree(TreeNode* root) {
        int maxdia = 0;

         helper(root,maxdia);
        return maxdia;
            }
};

File: 0543-diameter-of-binary-tree/0543-diameter-of-binary-tree.java

class Solution {
    public  int height(TreeNode root){
        if(root==null)  return 0;
        int l = height(root.left);
        int r = height(root.right);
        return Math.max(l,r)+1;
    }
    public int diameterOfBinaryTree(TreeNode root) {
        if(root==null)  return 0;
        int ld = diameterOfBinaryTree(root.left);
        int rd = diameterOfBinaryTree(root.right);
        int lsh = height(root.left);
        int rsh = height(root.right);
        int sd = lsh + rsh;
        return Math.max(sd,Math.max(ld,rd));
    }
}